server/client.go

```diff
@@ -1031,7 +1031,7 @@ func (c *client) readLoop(pre []byte) {
 		// Check if the account has mappings and if so set the local readcache flag.
 		// We check here to make sure any changes such as config reload are reflected here.
 		if c.kind == CLIENT {
-			if c.acc.hasMappings() {
+			if c.Account().hasMappings() {
```

Comment:
This is fine, but if we are concerned about acquiring the client lock just to check that, and since we are not protecting the whole "check+parse(which processes the message" anyway, we could also capture `acc` at top of function, under c's lock, and refresh at bottom of this function when we reacquire lock.

Reply:
That is a much better idea, did not remember the lock was re-acquired at the bottom, thanks!

Reply:
Will update.